### ISSUE TITLE
Fun with sockets, part 2

### ISSUE THREAD

**Issue by LunarWatcher** [LunarWatcher/stackchat.cpp#2]
This is an extension of #1, which I initially assumed was caused by the socket quietly  dying. After adding measures to revive the socket if it's closed, it's now clear that the problem is that it quietly dies on Stack's side, while still being kept alive. Effectively, the socket becomes a zombie socket.

I'm not clear on the cause, but neither is anyone else; chat is a piece of crap that hasn't been touched in around 10 years at this point. It's probably a good idea adding a way to detect if messages still appear.

During periods of inactivity, the socket still ends empty messages with no content, so checking if there have been any socket messages in the past X minutes, and restarting otherwise should do the trick.

**Comment by LunarWatcher** [comment]
This was fixed in 1dc32a8a4efef6a077d26b25af3e4bc92727f501. 

TL;DR: the socket quietly dies on Stack's side, and fails to disconnect or otherwise alert the socket. The client therefore has to make sure messages are received, and restart if those don't appear.

Empty messages (or occasionally one with some [garbage parameters](https://github.com/jbis9051/JamesSOBot/blob/master/docs/CHAT_API.md#basic-events)) are sent every 30 seconds, so a 5 minute wait is excessive, but ensures there isn't a misfire.

The patch has been verified to work for this particular type of socket death. One of the interesting artefacts with whatever this bug is, is that all sockets drop out at once. This also spreads across chat servers. No clue why, but the patch works and that's good enough for me
